llvm/test/CodeGen/AMDGPU/release-vgprs-gfx12.mir

```diff
@@ -0,0 +1,356 @@
+# RUN: llc -O2 -march=amdgcn -mcpu=gfx1200 -run-pass=si-insert-waitcnts -verify-machineinstrs -o - %s | FileCheck %s -check-prefixes=CHECK,DEFAULT
+# RUN: llc -O2 -march=amdgcn -mcpu=gfx1200 -mattr=+dynamic-vgpr -run-pass=si-insert-waitcnts -verify-machineinstrs -o - %s | FileCheck %s -check-prefixes=CHECK,DVGPR
```

Comment:
Don't need -O2

Reply:
Don't we? We only generate the S_ALLOC_VGPR when optimizing (there's one test where we don't because we put optnone on the function).

Reply:
-O2 is the default, and you are using run-pass anyway

Reply:
Oh, right. Fair point.